cmd/oras/blob/push.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package blob
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	digest "github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras/cmd/oras/internal/option"
+)
+
+type pushBlobOptions struct {
+	option.Common
+	option.Remote
+
+	FileRef   string
+	targetRef string
+}
+
+func pushBlob(opts pushBlobOptions) error {
+	ctx, _ := opts.SetLoggerLevel()
+	repo, err := opts.NewRepository(opts.targetRef, opts.Common)
+	if err != nil {
+		return err
+	}
+
+	// prepare blob content
+	desc, fp, err := packBlob(ctx, &opts)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		closeErr := fp.Close()
+		if err == nil {
+			err = closeErr
+		}
+	}()
+
+	// push blob
+	if err = repo.Push(ctx, desc, fp); err != nil {
+		return err
+	}
+
+	fmt.Println("Pushed", opts.targetRef)
+	fmt.Println("Digest:", desc.Digest)
+
+	return nil
+}
+
+func packBlob(ctx context.Context, opts *pushBlobOptions) (ocispec.Descriptor, *os.File, error) {
```

Comment:
Only opts.FileRef is used in this func, so maybe just pass a filename string is enough instead of pass the entire opts. I think about it because this way, I could reuse the same func for preparing manifest.

Reply:
I will move `func Prepare(path string, mediaType string) (ocispec.Descriptor, io.ReadCloser, error)` to the internal package so that it could be reused. thanks!